src/shared/loop-util.c

```diff
@@ -157,14 +158,29 @@ int loop_device_make(
 
 int loop_device_make_by_path(const char *path, int open_flags, uint32_t loop_flags, LoopDevice **ret) {
         _cleanup_close_ int fd = -1;
+        int r;
 
         assert(path);
         assert(ret);
-        assert(IN_SET(open_flags, O_RDWR, O_RDONLY));
+        assert(open_flags < 0 || IN_SET(open_flags, O_RDWR, O_RDONLY));
```

Comment:
The same change should be done for `loop_device_make()`, or make sure it never gets -1 for open_flags.

Reply:
nah, this function should patch open_flags to something valid if < 0. Fixed now.

Reply:
I can confirm that the assertion failure is gone after changing also `loop_device_make()` the same way.